sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/CryptographyClient.cs

```diff
@@ -160,7 +160,8 @@ internal CryptographyClient(Uri keyId, KeyVaultPipeline pipeline)
 
             _pipeline = pipeline;
             _remoteProvider = remoteClient;
-            _provider = remoteClient;
+
+            // Allow _provider to be set upon first request to the service depending on whether the key can be retrieved.
```

Comment:
I expected to see some other change related to this involving the setting of `_provider`. Should this comment go wherever that logic is as well?

Reply:
This comment was specifically to point out why `_provider` isn't set here.

Reply:
I know, but I was thinking that there must also be some other place that now deals with a potentially uninitialized `_provider` where some comment like this may be useful.

Reply:
There is - in `Initialize(Async)`. But I wanted to mention here why we weren't setting `_provider` in case people look at it and think, "huh, that's odd. I should add it in."